tests/test_pipeline_metrics.py

```
@@ -0,0 +1,103 @@
+        assert all(a.rule != "llm_silently_disabled" for a in anomalies)
+
+    def test_via_llm_does_not_fire(self):
+        ms = [self._metric(via="llm")]
+        anomalies = detect_anomalies(ms, has_llm_credentials=True)
+        assert all(a.rule != "llm_silently_disabled" for a in anomalies)
+
+    def test_produced_drift_fires_on_50pct_delta(self):
+        # 4 historical runs all around produced=30, latest = 1 (97% drop).
+        history = [self._metric(produced=p) for p in (30, 28, 32)]
+        latest = self._metric(produced=1)
+        anomalies = detect_anomalies(history + [latest], has_llm_credentials=False)
```

Ruff (0.15.12)

[warning] Consider `[*history, latest]` instead of concatenation

Replace with `[*history, latest]`

(RUF005)